src/Microsoft.DotNet.Cli.Utils/EnvironmentProvider.cs

```diff
@@ -54,6 +56,19 @@ private IEnumerable<string> SearchPaths
             }
         }
 
+        private string ExpandTildeSlash(string path)
+        {
+            string tildeSlash = "~/";
+            if (path.StartsWith(tildeSlash) && !string.IsNullOrEmpty(_userHomeDirectory.Value))
```

Comment:
`StringComparison.Ordinal` should be enough here for `StartsWith`

Reply:
How much performance gain we can get from it. `StringComparison.Ordinal` cannot convey the invention well

Reply:
Ordinal is a *lot* faster, but of course it depends on how often this is called, and so likely doesn't matter here.

I don't think performance is the main concern. I think semantically, we are not interested in CultureInfo.CurrentCulture affecting whether a path starts with something. If there is some culture where something other than precisely the same bytes as ~/ matches ~/, it would be a bug. File systems do not take current culture into account as string does by default.